pkg/csi_driver/controller.go

```diff
@@ -680,10 +681,24 @@ func getZoneFromSegment(seg map[string]string) (string, error) {
 	return zone, nil
 }
 
+func replaceInvalidChars(s string) (string, error) {
+	// Keys and values can contain letters, numbers, underscores and dashes:
+	// https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements,
+	reg, err := regexp.Compile(`[^\w-_]`)
+	if err != nil {
+		return "", fmt.Errorf("failed to compile regex to match valid labels: %w", err)
+	}
+	return reg.ReplaceAllString(s, "_"), nil
+}
+
 func extractLabels(parameters map[string]string, driverName string) (map[string]string, error) {
 	labels := make(map[string]string)
 	scLables := make(map[string]string)
 	for k, v := range parameters {
+		v, err := replaceInvalidChars(v)
```

Comment:
Ideally the driver should not take such decision to make intrinsic conversion of user provided labels.
We only make such conversion for the driver name since its not a user provided label.

I do see there is some conflict:
k8s naming [rules](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names) allow '.', while labels [rules](https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements) do not.

Can you point to which test causes this error?